main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -1671,8 +1684,8 @@ object Weeder extends Phase[ParsedAst.Program, WeededAst.Program] {
 
             argsVal map {
               case as =>
-                val lam = WeededAst.Expression.VarOrDefOrSig(law, loc)
-                val fun = WeededAst.Expression.VarOrDefOrSig(Name.QName(sp1, Name.RootNS, defn, sp2), loc)
+                val lam = WeededAst.Expression.Def(law, loc) // TODO allow signatures?
+                val fun = WeededAst.Expression.VarOrDefOrSig(defn, loc)
                 val exp = WeededAst.Expression.Apply(lam, fun :: as, loc)
                 WeededAst.Declaration.Property(law, defn, exp, loc)
```

Comment:
I think we are going to redesign this law thing soon, so this is probably not too important right now. You can keep the TODO.